pkg/middleware/middleware.go

```diff
@@ -128,7 +128,11 @@ func initContextWithApiKey(ctx *models.ReqContext) bool {
 	apikey := keyQuery.Result
 
 	// validate api key
-	if !apikeygen.IsValid(decoded, apikey.Key) {
+	isValid, err := apikeygen.IsValid(decoded, apikey.Key)
+	if err != nil {
+		ctx.JsonApiErr(500, "Validating API key failed", err)
```

Comment:
Not sure about this one, but shouldn't this return? By returning `false` it will continue to basic auth check here https://github.com/grafana/grafana/blob/0f4b970de81f5285da335cab64a2a7d249cc4c5c/pkg/middleware/middleware.go#L67. 

```suggestion
		ctx.JsonApiErr(500, "Validating API key failed", err)
		return false
```

Reply:
@marefr You're right that it should indeed return, it was an oversight from my side. However, doesn't it make sense to stop processing here and just return a 500 response to the client?

Reply:
Yeah should probably return `true` :+1:

Reply:
@marefr Implemented it.